homeassistant/components/withings/common.py

```diff
@@ -89,184 +370,278 @@ def __init__(
+        """Get the webhook config."""
+        return self._webhook_config
+
+    def _do_interval(
+        self, action: Callable[[], Coroutine], delay: timedelta, interval: timedelta
+    ) -> Callable:
+        cancel_func1 = None
+        cancel_func2 = None
+
+        @callback
+        def runner(now: datetime.datetime) -> None:
+            self._hass.async_run_job(action())
+
+        def cancel() -> None:
+            nonlocal cancel_func1
+            nonlocal cancel_func2
+            cancel_func1()
+            cancel_func2()
+
+        cancel_func1 = async_track_point_in_utc_time(
+            self._hass, action=runner, point_in_time=dt.utcnow() + delay,
+        )
 
-    @property
-    def measures(self) -> MeasureGetMeasResponse:
-        """Get the current measures data."""
-        return self._measures
+        cancel_func2 = async_track_time_interval(self._hass, runner, interval + delay)
 
-    @property
-    def sleep(self) -> SleepGetResponse:
-        """Get the current sleep data."""
-        return self._sleep
+        return cancel
 
-    @property
-    def sleep_summary(self) -> SleepGetSummaryResponse:
-        """Get the current sleep summary data."""
-        return self._sleep_summary
-
-    @staticmethod
-    def get_throttle_interval() -> int:
-        """Get the throttle interval."""
-        return const.THROTTLE_INTERVAL
-
-    def get_throttle_data(self, domain: str) -> ThrottleData:
-        """Get throttlel data."""
-        return self.throttle_data.get(domain)
-
-    def set_throttle_data(self, domain: str, throttle_data: ThrottleData):
-        """Set throttle data."""
-        self.throttle_data[domain] = throttle_data
-
-    @staticmethod
-    def print_service_unavailable() -> bool:
-        """Print the service is unavailable (once) to the log."""
-        if WithingsDataManager.service_available is not False:
-            _LOGGER.error("Looks like the service is not available at the moment")
-            WithingsDataManager.service_available = False
-            return True
+    def start_polling(self):
+        """Start polling for data.
 
-        return False
+        Starts two background tasks for updating sensors and configuring webhook
+        at withings (if enabled). Both processes include jiter to help reduce
+        load on the withings servers. The processes have a delayed start between
+        0 and 20 second. They repeat every 60 minutes.
+        """
+        self.stop_polling()
 
-    @staticmethod
-    def print_service_available() -> bool:
-        """Print the service is available (once) to to the log."""
-        if WithingsDataManager.service_available is not True:
-            _LOGGER.info("Looks like the service is available again")
-            WithingsDataManager.service_available = True
-            return True
+        self._cancel_interval_update_interval = self._do_interval(
+            action=lambda: self.async_update_all(),  # pylint: disable=unnecessary-lambda
```

Comment:
By making it a lambda, it is run in the executor so you are not allowed to call async functions.